bindings/pyroot_experimental/PyROOT/test/rdataframe_asnumpy.py

```diff
@@ -0,0 +1,199 @@
+import unittest
+import ROOT
+import numpy as np
+
+
+def make_tree(*dtypes):
+    tree = ROOT.TTree("test", "description")
+    col_names = ["col_{}".format(d) for d in dtypes]
+
+    col_vars = []
+    for dtype in dtypes:
+        if "F" in dtype:
+            var = np.empty(1, dtype=np.float32)
+        elif "D" in dtype:
+            var = np.empty(1, dtype=np.float64)
+        elif "I" in dtype:
+            var = np.empty(1, dtype=np.int32)
+        elif "i" in dtype:
+            var = np.empty(1, dtype=np.uint32)
+        elif "L" in dtype:
+            var = np.empty(1, dtype=np.int64)
+        elif "l" in dtype:
+            var = np.empty(1, dtype=np.uint64)
+        elif "S" in dtype:
+            var = np.empty(1, dtype=np.int16)
+        elif "s" in dtype:
+            var = np.empty(1, dtype=np.uint16)
+        elif "B" in dtype:
+            var = np.empty(1, dtype=np.int8)
+        elif "b" in dtype:
+            var = np.empty(1, dtype=np.uint8)
+        elif "O" in dtype:
+            var = np.empty(1, dtype=np.uint8)
+        else:
+            raise Exception(
+                "Type {} not known to create branch.".format(dtype))
+        col_vars.append(var)
+
+    for dtype, name, var in zip(dtypes, col_names, col_vars):
+        tree.Branch(name, var, name + "/" + dtype)
+
+    reference = {col: [] for col in col_names}
+    for i in range(5):
+        for i_var, var in enumerate(col_vars):
+            var[0] = i
+            reference[col_names[i_var]].append(var[0])
+        tree.Fill()
+    reference = {col: np.array(reference[col]) for col in reference}
+
+    return tree, reference, dtypes, col_names, col_vars
+
+
+def create_slice_in_scope():
+    df = ROOT.ROOT.RDataFrame(4).Define("x", "(double)rdfentry_")
+    npy = df.AsNumpy()
+    x = npy["x"]
+    x2 = x[:2]
+    return x2
+
+
+class RDataFrameAsNumpy(unittest.TestCase):
+    def test_branch_dtypes(self):
```

Comment:
Thank you @stwunsch , just one more thing: to be compliant with the rest of the pythonisation tests, can we add a bit of documentation to the test class to explain what the pythonisation is doing (and thus, what we are testing)? See e.g.
https://github.com/root-project/root/blob/master/bindings/pyroot_experimental/PyROOT/test/ttree_iterable.py#L10

Reply:
Done!